processor/metricstransformprocessor/metrics_transform_processor_helper.go

```diff
@@ -0,0 +1,372 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metricstransformprocessor
+
+import (
+	"fmt"
+	"math"
+	"math/rand"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	"github.com/golang/protobuf/ptypes/timestamp"
+)
+
+// timeseriesGroupByLabelValues is a data structure for grouping timeseries that will be aggregated
+type timeseriesGroupByLabelValues struct {
+	// key is a composite of the label values as a single string
+	// keyToTimeseriesMap groups timeseries by the label values
+	keyToTimeseriesMap map[string][]*metricspb.TimeSeries
+	// keyToLabelValuesMap maps the keys to the actual label values objects
+	keyToLabelValuesMap map[string][]*metricspb.LabelValue
+}
+
+// update updates the metric content based on operations indicated in transform.
+func (mtp *metricsTransformProcessor) update(metric *metricspb.Metric, transform Transform) {
```

Comment:
Is there a reason for using the opencensus format of metrics instead of the pdata? The pdata will more closely match the otel spec.

Reply:
I think when I created this pdata wasn't finalized. Is it finalized now? James and I have had this discussion before: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/347#discussion_r443991460. If I am updating this to use the new data format, it would definitely be a pretty big refactoring on all the code.

Reply:
Thank you for the comment link I couldn't find it earlier. Yes, the refactor would be a good exercise in patience and would be a separate pr.